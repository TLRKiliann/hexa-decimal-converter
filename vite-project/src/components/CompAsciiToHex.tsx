import React, { useState } from 'react'
import LabelInputComponent from './LabelInputComponent.js'
import ResultComponent from './ResultComponent.js'
import ButtonComponent from './ButtonComponent.js'

function CompAsciiToHex() {

    const [value, setValue] = useState<string>("");
    const [result, setResult] = useState<string | undefined>("");

    const nameValue = "ASCII";

    const converterTarget = (event: React.ChangeEvent<HTMLInputElement>): void => {
      const etv = event?.target.value;
      setValue(etv);
    }

    const convertResult = (): void => {
        let asciiCode = value.charCodeAt(0);
        let hexValue = asciiCode.toString(16);
        setResult("0x" + hexValue);
        setValue("");
    }

    return (
        <>
            <div className="labelinput--box">
                <LabelInputComponent 
                    value={value}
                    nameValue={nameValue}
                    converterTarget={(event)=>converterTarget(event)}
                >{nameValue}</LabelInputComponent>
            </div>

            <ButtonComponent convertResult={convertResult} />
            
            <div className="box--result">
                <ResultComponent 
                    result={result}
                />
            </div>
        </>
    )
}

export default CompAsciiToHex