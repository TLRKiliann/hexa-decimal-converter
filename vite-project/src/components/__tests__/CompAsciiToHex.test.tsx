import { test, expect } from "vitest"
import { create } from "react-test-renderer"
import CompAsciiToHex from '../CompAsciiToHex';

test('Snapshat of CompAsciiToHex', () => {
    const tree = create(<CompAsciiToHex />)
    expect(tree.toJSON()).toMatchSnapshot()
});