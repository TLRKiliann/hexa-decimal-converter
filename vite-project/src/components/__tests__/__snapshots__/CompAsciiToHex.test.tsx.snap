// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshat of CompAsciiToHex 1`] = `
[
  <div
    className="labelinput--box"
  >
    <label>
      ASCII
       :
    </label>
    <input
      onChange={[Function]}
      placeholder="ASCII"
      type="text"
      value=""
    />
  </div>,
  <button
    onClick={[Function]}
    type="button"
  >
    Convert
  </button>,
  <div
    className="box--result"
  >
    <p>
      Result : 
      Not binary !!!
    </p>
  </div>,
]
`;
